File: Day5/plusOne.java
class Solution {
    public int[] plusOne(int[] a) {

        int length=a.length-1;
        while(length>=0)
        
        {
            a[length]=a[length]+1;
            int carry=a[length]/10;
            if(carry==1)
            {
                a[length]=0;
                length=length-1;
            }
            else break;
        }
        if(a[0]==0)
        {
            int[] array=new int[a.length+1];
            array[0]=1;
            return array;
        }
        return a;
}
}

File: Day5/plusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        flag=0
        for i in digits :
            if i!=9 :
                flag=1
                break
        if flag==0 :
            l=[1]
            for i in range(len(digits)) :
                l.append(0)
            return l 
        else :
            c=1
            for i in range(len(digits)-1,-1,-1) :
                a=digits[i]+c
                if a>9 :
                    digits[i]=0
                    c=1
                else :
                    digits[i]=a
                    break
            return digits
